lib/hyperreduction/S_OPT.C

```diff
@@ -129,34 +131,75 @@ S_OPT(const Matrix* f_basis,
               false);
 
     RowInfo f_bv_max_local, f_bv_max_global;
-    f_bv_max_local.row_val = -1.0;
-    f_bv_max_local.proc = myid;
-    for (int i = 0; i < num_rows; ++i) {
-        double f_bv_val = fabs(Vo->item(i, 0));
-        if (f_bv_val > f_bv_max_local.row_val) {
-            f_bv_max_local.row_val = f_bv_val;
-            f_bv_max_local.row = i;
+
+    // Gather information about initial samples given as input.
+    const int num_init_samples = init_samples ? init_samples->size() : 0;
+    int total_num_init_samples = 0;
+    MPI_Allreduce(&num_init_samples, &total_num_init_samples, 1,
+                MPI_INT, MPI_SUM, MPI_COMM_WORLD);
+    CAROM_VERIFY(num_samples >= total_num_init_samples);
+
+    int init_sample_offset = 0;
+    for (int i = 0; i < total_num_init_samples; i++)
+    {
+        f_bv_max_local.row_val = -INT_MAX;
```

Comment:
`row_val` is a double, so this could be a large negative double. Using `INT_MAX` should work fine, but may be a little confusing.